src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -554,6 +554,62 @@ void LateInlineVirtualCallGenerator::do_late_inline() {
   CallGenerator::do_late_inline_helper();
 }
 
+static bool has_non_debug_usages(Node* n) {
+  for (DUIterator_Fast imax, i = n->fast_outs(imax); i < imax; i++) {
+    Node* m = n->fast_out(i);
+    if (!m->is_SafePoint()
+        || (m->is_Call() && m->as_Call()->has_non_debug_use(n))) {
+      return true;
+    }
+  }
+  return false;
+}
+
+static void scalarize_debug_usages(CallNode* call, Node* resproj) {
+  Unique_Node_List safepoints;
+  for (DUIterator_Fast imax, i = resproj->fast_outs(imax); i < imax; i++) {
+    Node* m = resproj->fast_out(i);
+    safepoints.push(m);
+  }
+
+#ifndef PRODUCT
+  if (PrintEliminateAllocations && safepoints.size() > 0) {
```

Comment:
`safepoints.size() > 0` is redundand.

Also, I'd move it to the end of the method after all the users are processed.